### DISCUSSION TITLE
INVALID_ID_INFORMATION error notify

### DISCUSSION THREAD

**Discussion by shahidge4** [strongswan/strongswan#1122] [Q&A] [answered]
Dear Team,
i am trying to establish ikev1 between ASA5550 and StrongSwan. but getting above error in phase 1.
 received stroke: initiate 'abc-to-xyz'
 initiating Main Mode IKE_SA abc-to-xyz[25] to a.b.c.d
 generating ID_PROT request 0 [ SA V V V V V ]
 sending packet: from w.x.y.z[500] to a.b.c.d[500] (180 bytes)
 received packet: from a.b.c.d[500] to w.x.y.z[500] (128 bytes)
 parsed ID_PROT response 0 [ SA V V ]
 received NAT-T (RFC 3947) vendor ID
 received FRAGMENTATION vendor ID
 generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
 sending packet: from w.x.y.z[500] to a.b.c.d[500] (244 bytes)
 sending retransmit 1 of request message ID 0, seq 2
 sending packet: from w.x.y.z[500] to a.b.c.d[500] (244 bytes)
 received packet: from a.b.c.d[500] to w.x.y.z[500] (304 bytes)
 parsed ID_PROT response 0 [ KE No V V V V NAT-D NAT-D ]
 received Cisco Unity vendor ID
 received XAuth vendor ID
 received unknown vendor ID: e4:40:7e:76:b7:1d:08:2f:06:87:62:8a:43:ba:f4:78
 received unknown vendor ID: 1f:07:f7:0e:aa:65:14:d3:b0:fa:96:54:2a:50:01:00
 generating ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
 sending packet: from w.x.y.z[500] to a.b.c.d[500] (108 bytes)
 received packet: from a.b.c.d[500] to w.x.y.z[500] (92 bytes)
 parsed ID_PROT response 0 [ ID HASH V ]
 received DPD vendor ID
 IKE_SA abc-to-xyz[25] established between w.x.y.z[w.x.y.z]...a.b.c.d[a.b.c.d]
 scheduling reauthentication in 27730s
 maximum IKE_SA lifetime 28270s
 generating QUICK_MODE request 3626665060 [ HASH SA No ID ID ]
 sending packet: from w.x.y.z[500] to a.b.c.d[500] (172 bytes)
 received packet: from a.b.c.d[500] to w.x.y.z[500] (220 bytes)
 parsed INFORMATIONAL_V1 request 1439180201 [ HASH N(INVAL_ID) ]
 received INVALID_ID_INFORMATION error notify
 received packet: from a.b.c.d[500] to w.x.y.z[500] (92 bytes)
 parsed INFORMATIONAL_V1 request 3467434943 [ HASH D ]
 received DELETE for IKE_SA abc-to-xyz[25]
 deleting IKE_SA abc-to-xyz[25] between w.x.y.z[w.x.y.z]...a.b.c.d[a.b.c.d]

and config file is 

config setup
        charondebug="all"
        uniqueids=yes
conn abc-to-xyz
        type=tunnel
        auto=start
        keyexchange=ikev1
        authby=secret
        left=w.x.y.z
        leftsubnet=192.168.9.209/32
        right=a.b.c.d
        rightsubnet=172.25.140.25/32
        ike=aes256-sha1-modp1024
        esp=aes256-sha1
        aggressive=yes
        keyingtries=%forever
        ikelifetime=28800s
        lifetime=3600s
        dpddelay=30s
        dpdtimeout=120s
        dpdaction=restart

**Comment by tobiasbrunner** [comment]
It's an error in phase 2 (Quick Mode), probably because `left|rightsubnet` are not correct.

**Reply by shahidge4** [reply]
thanks, can you help me to configure it. I am sharing a remote end-setting.
**Phase 1**

1. Authentication Method                             Pre-Shared Key
2. Encryption Scheme                                    IKEv1
3. Diffie-Hellman Group                                DH Group 2
4. Encryption Algorithm                                 AES-256
5. Hashing Algorithm                                     SHA
6. Main or Aggressive Mode                          Main
7. Lifetime (for renegotiation)                         28800s
8. Pre-Shared Key                                             ABC@xyz

**Phase 2**

1. Encapsulation (ESP or AH)                         ESP
2. Encryption Algorithm                                 AES-256
3. Authentication Algorithm                          SHA
4. Perfect Forward Secrecy                             yes
5. diffie-Hellman Group                                  DH Group 2
6. Lifetime in sec (for renegotiation)              86400s
7. Lifesize in KB (for renegotiation)                 4608000KB
8. remote private ips                                     172.25.140.25/32
9. local private ip                                           192.168.9.209/32

**ipsec.conf**
config setup
        # strictcrlpolicy=yes
        # uniqueids = no
        charondebug="ike 2, knl 2, cfg 2"
conn %default
        ikelifetime=86400s
        keylife=28800s
        rekeymargin=3m
        keyingtries=1
        keyexchange=ikev1
        authby=secret

conn ciscoios
        aggressive=no
        pfs=yes
        left=w.x.y.z                 #strongswan outside address
        leftsubnet=192.168.9.209/32         #network behind strongswan
        leftid=w.x.y.z               #IKEID sent by strongswan
        right=a.b.c.d                 # ASA outside address
        rightsubnet=172.25.140.25/32        #network behind ASA
        rightid=a.b.c.d               #IKEID sent by ASA
        auto=add
        ike=aes256-sha1-modp1024           #P1: modp1024 = DH group 2
        esp=aes256-sha1                   #P2
        leftauth=psk
        rightauth=psk
        keyexchange=ikev1

**Error/logs**
initiating Main Mode IKE_SA ciscoios[25] to 209.150.154.66
generating ID_PROT request 0 [ SA V V V V V ]
sending packet: from w.x.y.z[500] to a.b.c.d[500] (252 bytes)
received packet: from 2a.b.c.d[500] to w.x.y.z[500] (132 bytes)
parsed ID_PROT response 0 [ SA V V ]
received NAT-T (RFC 3947) vendor ID
received FRAGMENTATION vendor ID
generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
sending packet: from w.x.y.z[500] to a.b.c.d[500] (244 bytes)
received packet: from a.b.c.d[500] to w.x.y.z[500] (304 bytes)
parsed ID_PROT response 0 [ KE No V V V V NAT-D NAT-D ]
received Cisco Unity vendor ID
received XAuth vendor ID
received unknown vendor ID: 02:8f:fd:66:22:92:72:98:1a:62:75:7b:d2:a3:86:18
received unknown vendor ID: 1f:07:f7:0e:aa:65:14:d3:b0:fa:96:54:2a:50:01:00
generating ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
sending packet: from w.x.y.z[500] to a.b.c.d[500] (108 bytes)
received packet: from a.b.c.d[500] to w.x.y.z[500] (92 bytes)
parsed ID_PROT response 0 [ ID HASH V ]
received DPD vendor ID
IKE_SA ciscoios[25] established between w.x.y.z[95.216.28.243]...209.150.154.66[209.150.154.66]
scheduling reauthentication in 86103s
maximum IKE_SA lifetime 86283s
generating QUICK_MODE request 675277367 [ HASH SA No ID ID ]
sending packet: from w.x.y.z[500] to a.b.c.d[500] (204 bytes)
received packet: from a.b.c.d[500] to w.x.y.z[500] (92 bytes)
parsed INFORMATIONAL_V1 request 1757594404 [ HASH N((24576)) ]
received (24576) notify
received packet: from a.b.c.d[500] to w.x.y.z[500] (92 bytes)
parsed INFORMATIONAL_V1 request 35937131 [ HASH N(NO_PROP) ]
received NO_PROPOSAL_CHOSEN error notify
establishing connection 'ciscoios' failed

kindly help to resolve the above issue. I am trying for the last 3 days but not get any success.

**Reply by tobiasbrunner** [reply]
> diffie-Hellman Group DH Group 2

This is missing from your `esp` proposal.

**Comment by shahidge4** [comment, marked as answer]
Hi,
i added 
esp=aes256-sha1-modp1024

**Reply by shahidge4** [reply]
But same same error again
